src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -95,11 +96,15 @@ public List<AdvertisementTitleDto> getCreatedAdvertisements(@Parameter(hidden =
             @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
             @ApiResponse(responseCode = "403", description = "FORBIDDEN")})
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userChangePasswordDto,
-                                     @Parameter(hidden = true) Authentication authentication) throws InvalidDtoException {
-        User user = findUserByValidCredentials(authentication, userChangePasswordDto.getOldPassword());
+    public String updateUserPassword(@Valid ChangePasswordRequest changePasswordRequest,
+                                     @Parameter(hidden = true) Authentication authentication) throws DataConflictException {
+        User user = getUser(authentication.getName());
+        checkPasswordUniqueAndNotUsed(user.getPassword(),
+                changePasswordRequest.getPassword(), changePasswordRequest.getConfirmPassword());
+        user.setPassword(userService.updateUserPassword(changePasswordRequest.getPassword()));
+        userService.update(user);
```

Comment:
Didn't catch why do we need to save user twice?

Reply:
rename method in UserService